dotenv/mod.ts

```diff
@@ -1,187 +1,107 @@
-    }
-  }
+    if (groups.export != null) exports.add(groups.key);
 
-  return conf;
-}
+    let value = groups.value || "";
 
-function parseFile(filepath: string) {
-  try {
-    // Avoid errors that occur in deno deploy
-    // https://github.com/denoland/deno_std/issues/1957
-    if (typeof Deno.readFileSync !== "function") {
-      return {};
+    if (groups.quotes === `'`) {
+      // do nothing, preseve newlines
+    } else if (groups.quotes === `"`) {
+      value = value.replaceAll("\\n", "\n");
+    } else {
+      value = value.trim();
     }
-    return parse(new TextDecoder("utf-8").decode(Deno.readFileSync(filepath)));
-  } catch (e) {
-    if (e instanceof Deno.errors.NotFound) return {};
-    throw e;
-  }
-}
 
-async function parseFileAsync(filepath: string) {
-  try {
-    return parse(
-      new TextDecoder("utf-8").decode(await Deno.readFile(filepath)),
-    );
-  } catch (e) {
-    if (e instanceof Deno.errors.NotFound) return {};
-    throw e;
+    env[groups.key] = value;
   }
-}
-
-function isVariableStart(str: string): boolean {
-  return /^\s*[a-zA-Z_][a-zA-Z_0-9 ]*\s*=/.test(str);
-}
-
-function hasSingleQuotes(str: string): boolean {
-  return /^'([\s\S]*)'$/.test(str);
-}
 
-function hasDoubleQuotes(str: string): boolean {
-  return /^"([\s\S]*)"$/.test(str);
-}
+  const object = { env, exports: [...exports] };
+  verify(object, { allowEmptyValues, example });
 
-function expandNewlines(str: string): string {
-  return str.replaceAll("\\n", "\n");
+  return object;
 }
 
-function assertSafe(
-  conf: DotenvConfig,
-  confExample: DotenvConfig,
-  allowEmptyValues: boolean,
-) {
-  const currentEnv = Deno.env.toObject();
-
-  // Not all the variables have to be defined in .env, they can be supplied externally
-  const confWithEnv = Object.assign({}, currentEnv, conf);
-
-  const missing = difference(
-    Object.keys(confExample),
-    // If allowEmptyValues is false, filter out empty values from configuration
-    Object.keys(
-      allowEmptyValues ? confWithEnv : removeEmptyValues(confWithEnv),
-    ),
-  );
-
-  if (missing.length > 0) {
-    const errorMessages = [
-      `The following variables were defined in the example file but are not present in the environment:\n  ${
-        missing.join(
-          ", ",
-        )
-      }`,
-      `Make sure to add them to your env file.`,
-      !allowEmptyValues &&
-      `If you expect any of these variables to be empty, you can set the allowEmptyValues option to true.`,
-    ];
-
-    throw new MissingEnvVarsError(errorMessages.filter(Boolean).join("\n\n"));
-  }
-}
-
-export class MissingEnvVarsError extends Error {
-  constructor(message?: string) {
-    super(message);
-    this.name = "MissingEnvVarsError";
-    Object.setPrototypeOf(this, new.target.prototype);
-  }
-}
+// would be a nice to be able to generate .env files and complete parse/stringify pair like in JSON
+// export function stringify(object: DotEnvObject) {
+// }
```

Comment:
I'd very much like to see a variation on this functionality. See https://github.com/pietvanzoen/deno-dotenv/issues/42. I envision the use of `.env` as an accessible and simplistic file-based kvstore of string data.